### PR TITLE
allergies: update test

### PR DESCRIPTION
**Pull request by mcmillhj**
Update tests for the `allergies` exercise in response to #80

### CODE REVIEW

**Comment by snahor** [comment]
Please, don't include `README.md`.

**Comment by mcmillhj** [comment]
I thought all of the exercises had a `README.md` ?

**Comment by snahor** [comment]
You are right. But I had 2 reasons:

- I had created #70, becuase it would be an easy starting point for any new contributor.
- Not really sure about this one, but I think that your local `problem-specifications` is not up to date. I've noticed this after merging your previous PR. Actually, I started to suspect in one of your previous PRs.

I don't have any problem with you including `README.md`, just be sure that your local `problem-specifications` is updated.

File: exercises/allergies/allergies.sml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-datatype allergen = Eggs
+  datatype allergen = Eggs
```
Comment (snahor) [inline exercises/allergies/allergies.sml]: I think this is not needed.

File: exercises/allergies/test.sml [hunk 2]
```diff
@@ -1,132 +1,68 @@
...
+
+infixr |>
+fun x |> f = f x
+
+val testsuite =
+  describe "allergies" [
+    describe "allergicTo" [
+      describe "no allergies means not allergic" [
+        test "is not allergic to Peanuts"
+          (fn _ => allergicTo 0 Peanuts |> Expect.equalTo false),
```
Comment (snahor) [inline exercises/allergies/test.sml]: You could have also used `Expect.truthy` and `Expect.falsy`.

File: exercises/allergies/allergies.sml (hunk 1 above)
Comment (mcmillhj) [reply exercises/allergies/allergies.sml]: oh whoops, nice catch.

File: exercises/allergies/test.sml (hunk 2 above)
Comment (mcmillhj) [reply exercises/allergies/test.sml]: That sounds like it would be cleaner, I will update to use those two functions.

**Comment by mcmillhj** [comment]
@snahor definitely agree about the test cases being incorrect in canonical-data.json. I tried to raised the issue in the [exercism gitter](https://gitter.im/exercism/support) -- you might have to scroll up to see the conversation because I can't seem to anchor link to it -- but didn't really get anywhere. 

I'll file an issue with examples on the problem-descriptions repo

**Comment by mcmillhj** [comment]
@snahor also good point about the README, I will remove it from this PR.

**Comment by mcmillhj** [comment]
Hmm, I don't have a local problem-specifications repo. Should I?

**Comment by mcmillhj** [comment]
@snahor looks like the CI build expects a `README.md`, should I add an empty one?

**Comment by snahor** [comment]
Don't remove the `README.md`, when I told you not to include it I meant to not modify it :grin:. If you are using configlet to generate the `README.md` you need a local problem-spedifications. Check this https://github.com/exercism/sml#adding-an-exercise.

**Comment by mcmillhj** [comment]
Ah okay, I see what you mean now.

**Pull request merged**